pkg/vpcmodel/grouping.go

```diff
@@ -363,23 +326,24 @@ func (g *GroupConnLines) groupLinesByKey(srcGrouping, groupVsi bool) (res []*gro
 	groupingSrcOrDst = map[string][]*groupedConnLine{}
 	// populate map groupingSrcOrDst
 	for _, line := range g.GroupedLines {
-		srcOrDst, dstOrSrc := line.getSrcOrDst(srcGrouping), line.getSrcOrDst(!srcGrouping)
-		if !isInternalOfRequiredType(srcOrDst, groupVsi) {
+		grpTarget, grpIndex := line.getSrcOrDst(srcGrouping), line.getSrcOrDst(!srcGrouping)
+		if !isInternalOfRequiredType(grpTarget, groupVsi) {
 			res = append(res, line)
 			continue
 		}
-		key := getKeyOfGroupConnLines(dstOrSrc, line.commonProperties.groupingStrKey)
+		key := getKeyOfGroupConnLines(grpIndex, grpTarget, line.commonProperties.groupingStrKey)
 		if _, ok := groupingSrcOrDst[key]; !ok {
 			groupingSrcOrDst[key] = []*groupedConnLine{}
 		}
 		groupingSrcOrDst[key] = append(groupingSrcOrDst[key], line)
 	}
 	newGroupingSrcOrDst := g.extendGroupingSelfLoops(groupingSrcOrDst, srcGrouping)
+
 	return res, newGroupingSrcOrDst
 }
 
-func getKeyOfGroupConnLines(ep EndpointElem, connection string) string {
-	return ep.Name() + commaSeparator + connection
+func getKeyOfGroupConnLines(grpIndex, grpTarget EndpointElem, connection string) string {
+	return grpIndex.Name() + commaSeparator + connection + commaSeparator + getSubnetOrVPCUID(grpTarget)
```

Comment:
- change to `strings.Join()` 
- missing some documentation here - why adding `getSubnetOrVPCUID(grpTarget)` ? (explain that grouping is restricted to vsis of the same subnet and to subnets of the same vpc, and how this is related)
- should use `grpIndex.UID()` instead of name?

Reply:
`UID` is not implemented for endpointElement; should be implemented as part of #412
The other comments were inserted

Reply:
can you update the issue with description what is remaining to be handled?

Reply:
done